### ISSUE TITLE
Ignoring whitespace?

### ISSUE THREAD

**Issue by jbnv** [atom/github#1277]
I don't see an option in the settings to ignore whitespace. This is a major issue because beautifying and cleaning up code can cause changes in whitespace that we don't need to count as differences. In fact, counting whitespace changes as committable changes can make it difficult to see the real changes in the code?

Is there a way to ignore whitespace that I'm just not seeing?

**Comment by 50Wliu** [comment]
You're looking for the [atom/whitespace](https://github.com/atom/whitespace) package instead, which does contain settings to not cleanup trailing whitespace.

**Comment by jbnv** [comment]
No I'm not. I'm not talking just about trailing whitespace. I'm talking about indentation for one. I'm talking about a feature to make Git ignore whitespace when flagging something as a change.

It's this simple: Take a piece of code and increase its indent. I want to _not_ count that as a change to be committed. I don't see such a feature on the Github module. The whitespace module you mentioned doesn't say anything about giving Git the option to ignore whitespace.

Source Tree has this feature. I'm abandoning SourceTree for the Git modules because SourceTree is too slow and buggy. But this is a

**Comment by Arcanemagus** [comment]
For example, in GitKraken with whitespace showing in diffs you see this:
![image](https://user-images.githubusercontent.com/427137/34681889-16ffbe14-f452-11e7-872e-4ec142147682.png)

If the `Ignore white space` option is enabled, the diff shrinks to this:
![image](https://user-images.githubusercontent.com/427137/34681921-30bd6040-f452-11e7-9617-d741cc1f1bb8.png)

Note that there are several edge cases to consider here, such as when _all_ changes are hidden by the option:
![image](https://user-images.githubusercontent.com/427137/34681967-4a81557c-f452-11e7-8e54-437a9890a170.png)

Or when the whitespace changes are "important" (new lines?):
![image](https://user-images.githubusercontent.com/427137/34681999-62715e0c-f452-11e7-9ffd-dba5c29581c1.png)

These options are controlled in the `git` CLI tool with the common `--ignore-all-space` option, as well as the less common `--ignore-space-change` and `ignore-blank-lines`.

**Comment by BinaryMuse** [comment]
I agree, this would be a great addition, as long as the UX and staging experience works well with it.

**Comment by 50Wliu** [comment]
Ah, my apologies!

**Comment by mrthebob** [comment]
Oh boy, this would sure be useful, especially on very old code bases. I spend way too much time scrolling through stuff I don't want to commit.

**Comment by tryforceful** [comment]
+1. I'm actually surprised that this doesn't exist yet in this otherwise robust piece of software.

**Comment by Fess-AKA-DeadMonk** [comment]
is there any hope?

**Comment by utkarshgupta137** [comment]
I've made most of the required changes in my fork: https://github.com/utkarshgupta137/github/commit/4074685610927734cacbbe45a905b9525cd079c9.

The only thing left to do is move the option from package settings to this screen. If someone can help me with it, it would be great!

**Comment by utkarshgupta137** [comment]
A better way would be to highlight whitespace changes using a different/lighter color. You'll still have to scroll alot, but they would be easy to ignore.
